### PR TITLE
Support for LibreSSL

### PR DESCRIPTION
**Pull request by catatsuy**
Only libressl does not match option name and software name.

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -58,7 +62,14 @@ func (builder *Builder) name() string {
 }
 
 func (builder *Builder) option() string {
-	return fmt.Sprintf("--with-%s", builder.name())
+	name := builder.name()
+
+	// only libressl does not match option name
+	if name == "libressl" {
+		name = "openssl"
```
Comment (catatsuy) [inline builder/builder.go]: Only libressl does not match option name and software name. So I rewritten the option name here.

File: configure/generate.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ func Generate(configure string, modules3rd []module3rd.Module3rd, dependencies [
 
 	for _, d := range dependencies {
 		configure += fmt.Sprintf("%s=../%s-%s \\\n", d.Option, d.Name, d.Version)
-		if d.Name == "openssl" {
+		if d.Name == "openssl" || d.Name == "libressl" {
```
Comment (catatsuy) [inline configure/generate.go]: The options required by OpenSSL are also required for LibreSSL.

File: nginx-build.go [hunk 3]
```diff
@@ -188,6 +190,9 @@ func main() {
 	if *openResty && *tengine {
 		log.Fatal("select one between '-openresty' and '-tengine'.")
 	}
+	if *openSSLStatic && *libreSSLStatic {
```
Comment (catatsuy) [inline nginx-build.go]: Both `-openssl` and `-libressl` cannot be enabled at the same time.

**Comment by cubicdaiya** [comment]
LGTM. Could you add documents to README.md?

**Comment by catatsuy** [comment]
@cubicdaiya Thank you. I add documents for `-libressl` to README.md. Could you confirm it?

**Comment by cubicdaiya** [comment]
Merged. Thanks.

**Comment by catatsuy** [comment]
Thank you!

**Pull request merged**